bamt/utils/MathUtils.py

```diff
@@ -117,4 +122,107 @@ def component(data, columns, method):
                 biggest_p = p
                 comp_biggest = i
         n = comp_biggest
-    return n
\ No newline at end of file
+    return n
+
+
+def get_n_nearest(data, columns, corr=False, number_close=5):
+    """Returns N nearest neighbors for every column of dataframe, added into list
+
+    Args:
+        data (DataFrame): Proximity matrix
+        columns (list): df.columns.tolist()
+        corr (bool, optional): _description_. Defaults to False.
+        number_close (int, optional): Number of nearest neighbors. Defaults to 5.
+
+    Returns:
+        groups
+    """
+    groups = []
+    for c in columns:
+        if corr:
+            close_ind = data[c].sort_values(ascending=False).index.tolist()
+        else:
+            close_ind = data[c].sort_values().index.tolist()
+        groups.append(close_ind[0:number_close + 1])
+
+    return groups
+
+
+def get_proximity_matrix(df, df_coded, proximity_metric) -> pd.DataFrame:
```

Comment:
I don't like that the function accepts two kinds of data - source and encoded. I propose to make it so that it accepts only source ones and already inside, if encoding is needed, it produces encoding.

Reply:
However I've fixed it